clients/vault/src/oracle/collector/proof_builder.rs

```diff
@@ -243,61 +237,66 @@ impl ScpMessageCollector {
 					}
 				});
 
-				if let Some(i) = value {
-					if let ScpHistoryEntry::V0(scp_entry_v0) = i {
-						let slot_scp_envelopes = scp_entry_v0.clone().ledger_messages.messages;
-						let vec_scp = slot_scp_envelopes.get_vec().clone();
-
-						// Filter out any envelopes that are not externalize or confirm statements
-						let relevant_envelopes = vec_scp
-							.into_iter()
-							.filter(|scp| match scp.statement.pledges {
-								ScpStatementPledges::ScpStExternalize(_) |
-								ScpStatementPledges::ScpStConfirm(_) => true,
-								_ => false,
-							})
-							.collect::<Vec<_>>();
-
-						let externalized_envelopes_count = relevant_envelopes
-							.iter()
-							.filter(|scp| match scp.statement.pledges {
-								ScpStatementPledges::ScpStExternalize(_) => true,
-								_ => false,
-							})
-							.count();
-
-						// Ensure that at least one envelope is externalized
-						if externalized_envelopes_count == 0 {
-							tracing::error!(
+				let Some(ScpHistoryEntry::V0(scp_entry_v0)) = value else {
+					tracing::warn!("get_envelopes_from_horizon_archive(): Could not get ScpHistory entry from archive {archive_url} for slot {slot}");
+					continue;
+				};
+
+				let slot_scp_envelopes = scp_entry_v0.clone().ledger_messages.messages;
+				let vec_scp = slot_scp_envelopes.get_vec().clone();
+
+				// Filter out any envelopes that are not externalize or confirm statements
+				let mut relevant_envelopes = vec_scp
+					.into_iter()
+					.filter(|scp| match scp.statement.pledges {
+						ScpStatementPledges::ScpStExternalize(_) |
+						ScpStatementPledges::ScpStConfirm(_) => true,
+						_ => false,
+					})
+					.collect::<Vec<_>>();
+
+				let externalized_envelopes_count = relevant_envelopes
+					.iter()
+					.filter(|scp| match scp.statement.pledges {
+						ScpStatementPledges::ScpStExternalize(_) => true,
+						_ => false,
+					})
+					.count();
+
+				// Ensure that at least one envelope is externalized
+				if externalized_envelopes_count == 0 {
+					tracing::error!(
 							"get_envelopes_from_horizon_archive(): The contained archive entry fetched from {} for slot {slot} is invalid because it does not contain any externalized envelopes.",
 								scp_archive_storage.0
 						);
-							// remove the file since it's invalid.
-							scp_archive_storage.remove_file(slot);
-							continue;
-						}
-
-						let mut envelopes_map = envelopes_map_arc.write();
-						let mut from_archive_map = env_from_archive_map.write();
-
-						if envelopes_map.get(&slot).is_none() {
```

Comment:
This caused the [timeout issue](https://github.com/pendulum-chain/spacewalk/actions/runs/11176052005/job/31083875194#step:4:5645).

The map had 1 envelope  for the given slot. Since it did not return `none`, this insert logic was always skipped.

I appended the existing envelope wit the envelopes found in the archive. See lines 287-292.

Reply:
Wow, very nice catch!

Reply:
That must have been hard to catch 😅. What was the reason we were checking for an empty map in the first place? And why we don't care now to append.

Reply:
> What was the reason we were checking for an empty map in the first place? 

@gianfra-t  good qustion, and I don't have an answer. :see_no_evil:


> And why we don't care now to append.

We don't append anymore because of possible dupes. See [Marcel's comment](https://github.com/pendulum-chain/spacewalk/pull/545#discussion_r1791591783):
   _...
   if we send duplicates to the oracle, this also increases the size of the already large extrinsic call. 
   ..._

Reply:
Great 👍. I miss that other comment. Thanks!